⠧ Running Flake8...
⠦ Running PyLint...
⠹ Running MyPy...
                                                                                     Jack-O-Linter Results                                                                                     
┏━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Tool   ┃  Score ┃ Errors                                                                              ┃ Warnings                                                                            ┃
┡━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ MyPy   │  98.93 │ src/requests/compat.py:64: error: Name "json" already defined (by an import)        │ src/requests/cookies.py:20: note: See                                               │
│        │        │ src/requests/compat.py:69: error: Incompatible import of "JSONDecodeError"          │ https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports             │
│        │        │ (imported name has type "type", local name has type "type")                         │ src/requests/help.py:31: note: Hint: "python3 -m pip install types-pyOpenSSL"       │
│        │        │ src/requests/compat.py:93: error: Module "urllib.request" has no attribute          │ src/requests/help.py:31: note: (or run "mypy --install-types" to install all        │
│        │        │ "proxy_bypass_environment"                                                          │ missing stub packages)                                                              │
│        │        │ src/requests/cookies.py:20: error: Cannot find implementation or library stub for   │                                                                                     │
│        │        │ module named "dummy_threading"                                                      │                                                                                     │
│        │        │ src/requests/cookies.py:20: error: Name "threading" already defined (by an import)  │                                                                                     │
│        │        │ src/requests/models.py:72: error: "LookupDict" has no attribute "moved"             │                                                                                     │
│        │        │ src/requests/models.py:73: error: "LookupDict" has no attribute "found"             │                                                                                     │
│        │        │ src/requests/models.py:74: error: "LookupDict" has no attribute "other"             │                                                                                     │
│        │        │ src/requests/models.py:75: error: "LookupDict" has no attribute                     │                                                                                     │
│        │        │ "temporary_redirect"                                                                │                                                                                     │
│        │        │ src/requests/models.py:76: error: "LookupDict" has no attribute                     │                                                                                     │
│        │        │ "permanent_redirect"                                                                │                                                                                     │
│        │        │ src/requests/adapters.py:63: error: Name "SOCKSProxyManager" already defined        │                                                                                     │
│        │        │ (possibly by an import)                                                             │                                                                                     │
│        │        │ src/requests/adapters.py:87: error: Incompatible types in assignment (expression    │                                                                                     │
│        │        │ has type "None", variable has type "SSLContext")                                    │                                                                                     │
│        │        │ src/requests/adapters.py:95: error: Invalid type comment or annotation              │                                                                                     │
│        │        │ src/requests/adapters.py:117: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "str", target has type "SSLContext")                                       │                                                                                     │
│        │        │ src/requests/adapters.py:119: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "str", target has type "SSLContext")                                       │                                                                                     │
│        │        │ src/requests/adapters.py:120: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "str", target has type "SSLContext")                                       │                                                                                     │
│        │        │ src/requests/adapters.py:123: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "str", target has type "SSLContext")                                       │                                                                                     │
│        │        │ src/requests/adapters.py:124: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "str", target has type "SSLContext")                                       │                                                                                     │
│        │        │ src/requests/adapters.py:128: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "str", target has type "SSLContext")                                       │                                                                                     │
│        │        │ src/requests/__init__.py:50: error: Incompatible types in assignment (expression    │                                                                                     │
│        │        │ has type "None", variable has type "str")                                           │                                                                                     │
│        │        │ src/requests/__init__.py:55: error: Incompatible types in assignment (expression    │                                                                                     │
│        │        │ has type "None", variable has type "str")                                           │                                                                                     │
│        │        │ src/requests/__init__.py:128: error: Incompatible types in assignment (expression   │                                                                                     │
│        │        │ has type "None", variable has type Module)                                          │                                                                                     │
│        │        │ src/requests/help.py:16: error: Incompatible types in assignment (expression has    │                                                                                     │
│        │        │ type "None", variable has type Module)                                              │                                                                                     │
│        │        │ src/requests/help.py:21: error: Incompatible types in assignment (expression has    │                                                                                     │
│        │        │ type "None", variable has type Module)                                              │                                                                                     │
│        │        │ src/requests/help.py:26: error: Incompatible types in assignment (expression has    │                                                                                     │
│        │        │ type "None", variable has type Module)                                              │                                                                                     │
│        │        │ src/requests/help.py:30: error: Incompatible import of "cryptography" (imported     │                                                                                     │
│        │        │ name has type Module, local name has type "object")                                 │                                                                                     │
│        │        │ src/requests/help.py:31: error: Library stubs not installed for "OpenSSL"           │                                                                                     │
│        │        │ src/requests/help.py:31: error: Name "OpenSSL" already defined on line 27           │                                                                                     │
├────────┼────────┼─────────────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────────────┤
│ Flake8 │ 100.00 │ No Errors                                                                           │ No Warnings                                                                         │
├────────┼────────┼─────────────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────────────┤
│ PyLint │  70.83 │ src/requests/__init__.py:58:0: C0116: Missing function or method docstring          │ src/requests/__init__.py:165:0: W0622: Redefining built-in 'ConnectionError'        │
│        │        │ (missing-function-docstring)                                                        │ (redefined-builtin)                                                                 │
│        │        │ src/requests/__init__.py:101:18: C0209: Formatting a regular string which could be  │ src/requests/__init__.py:165:0: W0622: Redefining built-in 'ConnectionError'        │
│        │        │ an f-string (consider-using-f-string)                                               │ (redefined-builtin)                                                                 │
│        │        │ src/requests/__init__.py:114:8: C0209: Formatting a regular string which could be   │ src/requests/__init__.py:58:41: W0621: Redefining name 'chardet_version' from outer │
│        │        │ an f-string (consider-using-f-string)                                               │ scope (line 53) (redefined-outer-name)                                              │
│        │        │ src/requests/__init__.py:143:0: C0413: Import "from urllib3.exceptions import       │ src/requests/__init__.py:58:58: W0621: Redefining name 'charset_normalizer_version' │
│        │        │ DependencyWarning" should be placed at the top of the module                        │ from outer scope (line 48) (redefined-outer-name)                                   │
│        │        │ (wrong-import-position)                                                             │ src/requests/__init__.py:67:18: W0621: Redefining name 'patch' from outer scope     │
│        │        │ src/requests/__init__.py:143:0: C0413: Import "from urllib3.exceptions import       │ (line 164) (redefined-outer-name)                                                   │
│        │        │ DependencyWarning" should be placed at the top of the module                        │ src/requests/__init__.py:58:41: W0621: Redefining name 'chardet_version' from outer │
│        │        │ (wrong-import-position)                                                             │ scope (line 53) (redefined-outer-name)                                              │
│        │        │ src/requests/__init__.py:148:0: C0413: Import "import logging" should be placed at  │ src/requests/__init__.py:58:58: W0621: Redefining name 'charset_normalizer_version' │
│        │        │ the top of the module (wrong-import-position)                                       │ from outer scope (line 48) (redefined-outer-name)                                   │
│        │        │ src/requests/__init__.py:148:0: C0413: Import "import logging" should be placed at  │ src/requests/__init__.py:67:18: W0621: Redefining name 'patch' from outer scope     │
│        │        │ the top of the module (wrong-import-position)                                       │ (line 164) (redefined-outer-name)                                                   │
│        │        │ src/requests/__init__.py:149:0: C0413: Import "from logging import NullHandler"     │ src/requests/__init__.py:93:24: W0621: Redefining name 'cryptography_version' from  │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ outer scope (line 136) (redefined-outer-name)                                       │
│        │        │ src/requests/__init__.py:149:0: C0413: Import "from logging import NullHandler"     │ src/requests/__init__.py:93:24: W0621: Redefining name 'cryptography_version' from  │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ outer scope (line 136) (redefined-outer-name)                                       │
│        │        │ src/requests/__init__.py:151:0: C0413: Import "from . import packages, utils"       │ src/requests/cookies.py:131:56: W0212: Access to a protected member                 │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ _original_response of a client class (protected-access)                             │
│        │        │ src/requests/__init__.py:151:0: C0413: Import "from . import packages, utils"       │ src/requests/cookies.py:136:23: W0212: Access to a protected member                 │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ _original_response of a client class (protected-access)                             │
│        │        │ src/requests/__init__.py:152:0: C0413: Import "from .__version__ import __author__, │ src/requests/cookies.py:194:4: W0237: Parameter 'key' has been renamed to 'name' in │
│        │        │ __author_email__, __build__, __cake__, __copyright__, __description__, __license__, │ overriding 'RequestsCookieJar.get' method (arguments-renamed)                       │
│        │        │ __title__, __url__, __version__" should be placed at the top of the module          │ src/requests/cookies.py:358:4: W0221: Number of parameters was 1 in                 │
│        │        │ (wrong-import-position)                                                             │ 'MutableMapping.update' and is now 2 in overriding 'RequestsCookieJar.update'       │
│        │        │ src/requests/__init__.py:152:0: C0413: Import "from .__version__ import __author__, │ method (arguments-differ)                                                           │
│        │        │ __author_email__, __build__, __cake__, __copyright__, __description__, __license__, │ src/requests/cookies.py:358:4: W0221: Variadics removed in overriding               │
│        │        │ __title__, __url__, __version__" should be placed at the top of the module          │ 'RequestsCookieJar.update' method (arguments-differ)                                │
│        │        │ (wrong-import-position)                                                             │ src/requests/cookies.py:500:12: W0707: Consider explicitly re-raising using 'except │
│        │        │ src/requests/__init__.py:164:0: C0413: Import "from .api import delete, get, head,  │ ValueError as exc' and 'raise TypeError(f"max-age: {morsel['max-age']} must be      │
│        │        │ options, patch, post, put, request" should be placed at the top of the module       │ integer") from exc' (raise-missing-from)                                            │
│        │        │ (wrong-import-position)                                                             │ src/requests/cookies.py:500:12: W0707: Consider explicitly re-raising using 'except │
│        │        │ src/requests/__init__.py:164:0: C0413: Import "from .api import delete, get, head,  │ ValueError as exc' and 'raise TypeError(f"max-age: {morsel['max-age']} must be      │
│        │        │ options, patch, post, put, request" should be placed at the top of the module       │ integer") from exc' (raise-missing-from)                                            │
│        │        │ (wrong-import-position)                                                             │ src/requests/auth.py:181:9: W0511: XXX not implemented yet (fixme)                  │
│        │        │ src/requests/__init__.py:165:0: C0413: Import "from .exceptions import              │ src/requests/auth.py:215:13: W0511: XXX handle auth-int. (fixme)                    │
│        │        │ ConnectionError, ConnectTimeout, FileModeWarning, HTTPError, JSONDecodeError,       │ src/requests/auth.py:220:9: W0511: XXX should the partial digests be encoded too?   │
│        │        │ ReadTimeout, RequestException, Timeout, TooManyRedirects, URLRequired" should be    │ (fixme)                                                                             │
│        │        │ placed at the top of the module (wrong-import-position)                             │ src/requests/auth.py:181:9: W0511: XXX not implemented yet (fixme)                  │
│        │        │ src/requests/__init__.py:165:0: C0413: Import "from .exceptions import              │ src/requests/auth.py:215:13: W0511: XXX handle auth-int. (fixme)                    │
│        │        │ ConnectionError, ConnectTimeout, FileModeWarning, HTTPError, JSONDecodeError,       │ src/requests/auth.py:220:9: W0511: XXX should the partial digests be encoded too?   │
│        │        │ ReadTimeout, RequestException, Timeout, TooManyRedirects, URLRequired" should be    │ (fixme)                                                                             │
│        │        │ placed at the top of the module (wrong-import-position)                             │ src/requests/auth.py:17:0: W0622: Redefining built-in 'str' (redefined-builtin)     │
│        │        │ src/requests/__init__.py:177:0: C0413: Import "from .models import PreparedRequest, │ src/requests/auth.py:17:0: W0622: Redefining built-in 'str' (redefined-builtin)     │
│        │        │ Request, Response" should be placed at the top of the module                        │ src/requests/auth.py:236:0: W0613: Unused argument 'kwargs' (unused-argument)       │
│        │        │ (wrong-import-position)                                                             │ src/requests/auth.py:236:0: W0613: Unused argument 'kwargs' (unused-argument)       │
│        │        │ src/requests/__init__.py:177:0: C0413: Import "from .models import PreparedRequest, │ src/requests/auth.py:267:12: W0104: Statement seems to have no effect               │
│        │        │ Request, Response" should be placed at the top of the module                        │ (pointless-statement)                                                               │
│        │        │ (wrong-import-position)                                                             │ src/requests/auth.py:270:35: W0212: Access to a protected member _cookies of a      │
│        │        │ src/requests/__init__.py:178:0: C0413: Import "from .sessions import Session,       │ client class (protected-access)                                                     │
│        │        │ session" should be placed at the top of the module (wrong-import-position)          │ src/requests/auth.py:271:33: W0212: Access to a protected member _cookies of a      │
│        │        │ src/requests/__init__.py:178:0: C0413: Import "from .sessions import Session,       │ client class (protected-access)                                                     │
│        │        │ session" should be placed at the top of the module (wrong-import-position)          │ src/requests/sessions.py:240:35: W0212: Access to a protected member _cookies of a  │
│        │        │ src/requests/__init__.py:179:0: C0413: Import "from .status_codes import codes"     │ client class (protected-access)                                                     │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ src/requests/sessions.py:241:26: W0212: Access to a protected member _cookies of a  │
│        │        │ src/requests/__init__.py:179:0: C0413: Import "from .status_codes import codes"     │ client class (protected-access)                                                     │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ src/requests/sessions.py:242:45: W0212: Access to a protected member _cookies of a  │
│        │        │ src/requests/__init__.py:143:0: C0411: third party import                           │ client class (protected-access)                                                     │
│        │        │ "urllib3.exceptions.DependencyWarning" should be placed before local import         │ src/requests/sessions.py:251:25: W0212: Access to a protected member _body_position │
│        │        │ "exceptions.RequestsDependencyWarning" (wrong-import-order)                         │ of a client class (protected-access)                                                │
│        │        │ src/requests/__init__.py:148:0: C0411: standard import "logging" should be placed   │ src/requests/sessions.py:739:16: W0212: Access to a protected member _next of a     │
│        │        │ before third party imports "urllib3", "urllib3.exceptions.DependencyWarning" and    │ client class (protected-access)                                                     │
│        │        │ local import "exceptions.RequestsDependencyWarning" (wrong-import-order)            │ src/requests/sessions.py:746:12: W0104: Statement seems to have no effect           │
│        │        │ src/requests/__init__.py:149:0: C0411: standard import "logging.NullHandler" should │ (pointless-statement)                                                               │
│        │        │ be placed before third party imports "urllib3",                                     │ src/requests/sessions.py:33:0: W0611: Unused REDIRECT_STATI imported from models    │
│        │        │ "urllib3.exceptions.DependencyWarning" and local import                             │ (unused-import)                                                                     │
│        │        │ "exceptions.RequestsDependencyWarning" (wrong-import-order)                         │ src/requests/sessions.py:41:0: W0611: Unused should_bypass_proxies imported from    │
│        │        │ src/requests/__init__.py:143:0: C0412: Imports from package urllib3 are not grouped │ utils (unused-import)                                                               │
│        │        │ (ungrouped-imports)                                                                 │ src/requests/sessions.py:33:0: W0611: Unused REDIRECT_STATI imported from models    │
│        │        │ src/requests/__init__.py:143:0: C0411: third party import                           │ (unused-import)                                                                     │
│        │        │ "urllib3.exceptions.DependencyWarning" should be placed before local import         │ src/requests/sessions.py:41:0: W0611: Unused should_bypass_proxies imported from    │
│        │        │ "exceptions.RequestsDependencyWarning" (wrong-import-order)                         │ utils (unused-import)                                                               │
│        │        │ src/requests/__init__.py:148:0: C0411: standard import "logging" should be placed   │ src/requests/hooks.py:19:1: W0511: TODO: response is the only one (fixme)           │
│        │        │ before third party imports "urllib3", "urllib3.exceptions.DependencyWarning" and    │ src/requests/hooks.py:19:1: W0511: TODO: response is the only one (fixme)           │
│        │        │ local import "exceptions.RequestsDependencyWarning" (wrong-import-order)            │ src/requests/compat.py:102:0: W0622: Redefining built-in 'str' (redefined-builtin)  │
│        │        │ src/requests/__init__.py:149:0: C0411: standard import "logging.NullHandler" should │ src/requests/compat.py:103:0: W0622: Redefining built-in 'bytes'                    │
│        │        │ be placed before third party imports "urllib3",                                     │ (redefined-builtin)                                                                 │
│        │        │ "urllib3.exceptions.DependencyWarning" and local import                             │ src/requests/compat.py:102:0: W0622: Redefining built-in 'str' (redefined-builtin)  │
│        │        │ "exceptions.RequestsDependencyWarning" (wrong-import-order)                         │ src/requests/compat.py:103:0: W0622: Redefining built-in 'bytes'                    │
│        │        │ src/requests/__init__.py:143:0: C0412: Imports from package urllib3 are not grouped │ (redefined-builtin)                                                                 │
│        │        │ (ungrouped-imports)                                                                 │ src/requests/compat.py:32:4: W0621: Redefining name 'chardet' from outer scope      │
│        │        │ src/requests/cookies.py:40:4: C0116: Missing function or method docstring           │ (line 42) (redefined-outer-name)                                                    │
│        │        │ (missing-function-docstring)                                                        │ src/requests/compat.py:32:4: W0621: Redefining name 'chardet' from outer scope      │
│        │        │ src/requests/cookies.py:43:4: C0116: Missing function or method docstring           │ (line 42) (redefined-outer-name)                                                    │
│        │        │ (missing-function-docstring)                                                        │ src/requests/compat.py:102:0: W0127: Assigning the same variable 'str' to itself    │
│        │        │ src/requests/cookies.py:46:4: C0116: Missing function or method docstring           │ (self-assigning-variable)                                                           │
│        │        │ (missing-function-docstring)                                                        │ src/requests/compat.py:103:0: W0127: Assigning the same variable 'bytes' to itself  │
│        │        │ src/requests/cookies.py:49:4: C0116: Missing function or method docstring           │ (self-assigning-variable)                                                           │
│        │        │ (missing-function-docstring)                                                        │ src/requests/compat.py:60:4: W0611: Unused simplejson imported as json              │
│        │        │ src/requests/cookies.py:69:4: C0116: Missing function or method docstring           │ (unused-import)                                                                     │
│        │        │ (missing-function-docstring)                                                        │ src/requests/compat.py:64:4: W0611: Unused import json (unused-import)              │
│        │        │ src/requests/cookies.py:72:4: C0116: Missing function or method docstring           │ src/requests/compat.py:67:4: W0611: Unused JSONDecodeError imported from simplejson │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:75:4: C0116: Missing function or method docstring           │ src/requests/compat.py:72:0: W0611: Unused OrderedDict imported from collections    │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:84:4: C0116: Missing function or method docstring           │ src/requests/compat.py:73:0: W0611: Unused Callable imported from collections.abc   │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:87:4: C0116: Missing function or method docstring           │ src/requests/compat.py:73:0: W0611: Unused Mapping imported from collections.abc    │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:91:4: C0116: Missing function or method docstring           │ src/requests/compat.py:73:0: W0611: Unused MutableMapping imported from             │
│        │        │ (missing-function-docstring)                                                        │ collections.abc (unused-import)                                                     │
│        │        │ src/requests/cookies.py:95:4: C0116: Missing function or method docstring           │ src/requests/compat.py:74:0: W0611: Unused cookiejar imported from http as          │
│        │        │ (missing-function-docstring)                                                        │ cookielib (unused-import)                                                           │
│        │        │ src/requests/cookies.py:99:4: C0116: Missing function or method docstring           │ src/requests/compat.py:75:0: W0611: Unused Morsel imported from http.cookies        │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:117:4: C0116: Missing function or method docstring          │ src/requests/compat.py:76:0: W0611: Unused StringIO imported from io                │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:120:4: C0116: Missing function or method docstring          │ src/requests/compat.py:81:0: W0611: Unused quote imported from urllib.parse         │
│        │        │ (missing-function-docstring)                                                        │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:166:8: R1704: Redefining argument with the local name       │ src/requests/compat.py:81:0: W0611: Unused quote_plus imported from urllib.parse    │
│        │        │ 'domain' (redefined-argument-from-local)                                            │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:166:16: R1704: Redefining argument with the local name      │ src/requests/compat.py:81:0: W0611: Unused unquote imported from urllib.parse       │
│        │        │ 'path' (redefined-argument-from-local)                                              │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:166:22: R1704: Redefining argument with the local name      │ src/requests/compat.py:81:0: W0611: Unused unquote_plus imported from urllib.parse  │
│        │        │ 'name' (redefined-argument-from-local)                                              │ (unused-import)                                                                     │
│        │        │ src/requests/cookies.py:206:4: R1710: Either all return statements in a function    │ src/requests/compat.py:81:0: W0611: Unused urldefrag imported from urllib.parse     │
│        │        │ should return an expression, or none of them should.                                │ (unused-import)                                                                     │
│        │        │ (inconsistent-return-statements)                                                    │ src/requests/compat.py:81:0: W0611: Unused urlencode imported from urllib.parse     │
│        │        │ src/requests/cookies.py:398:8: C0103: Variable name "toReturn" doesn't conform to   │ (unused-import)                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │ src/requests/compat.py:81:0: W0611: Unused urljoin imported from urllib.parse       │
│        │        │ src/requests/cookies.py:409:24: C0103: Variable name "toReturn" doesn't conform to  │ (unused-import)                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │ src/requests/compat.py:81:0: W0611: Unused urlparse imported from urllib.parse      │
│        │        │ src/requests/auth.py:37:12: C0209: Formatting a regular string which could be an    │ (unused-import)                                                                     │
│        │        │ f-string (consider-using-f-string)                                                  │ src/requests/compat.py:81:0: W0611: Unused urlsplit imported from urllib.parse      │
│        │        │ src/requests/auth.py:47:12: C0209: Formatting a regular string which could be an    │ (unused-import)                                                                     │
│        │        │ f-string (consider-using-f-string)                                                  │ src/requests/compat.py:81:0: W0611: Unused urlunparse imported from urllib.parse    │
│        │        │ src/requests/auth.py:69:0: R0903: Too few public methods (1/2)                      │ (unused-import)                                                                     │
│        │        │ (too-few-public-methods)                                                            │ src/requests/compat.py:93:0: W0611: Unused getproxies imported from urllib.request  │
│        │        │ src/requests/auth.py:69:0: R0903: Too few public methods (1/2)                      │ (unused-import)                                                                     │
│        │        │ (too-few-public-methods)                                                            │ src/requests/compat.py:93:0: W0611: Unused getproxies_environment imported from     │
│        │        │ src/requests/auth.py:99:0: R0903: Too few public methods (1/2)                      │ urllib.request (unused-import)                                                      │
│        │        │ (too-few-public-methods)                                                            │ src/requests/compat.py:93:0: W0611: Unused parse_http_list imported from            │
│        │        │ src/requests/auth.py:99:0: R0903: Too few public methods (1/2)                      │ urllib.request (unused-import)                                                      │
│        │        │ (too-few-public-methods)                                                            │ src/requests/compat.py:93:0: W0611: Unused proxy_bypass imported from               │
│        │        │ src/requests/auth.py:116:4: C0116: Missing function or method docstring             │ urllib.request (unused-import)                                                      │
│        │        │ (missing-function-docstring)                                                        │ src/requests/compat.py:93:0: W0611: Unused proxy_bypass_environment imported from   │
│        │        │ src/requests/auth.py:126:4: R0914: Too many local variables (28/15)                 │ urllib.request (unused-import)                                                      │
│        │        │ (too-many-locals)                                                                   │ src/requests/compat.py:60:4: W0611: Unused simplejson imported as json              │
│        │        │ src/requests/auth.py:126:4: R0914: Too many local variables (28/15)                 │ (unused-import)                                                                     │
│        │        │ (too-many-locals)                                                                   │ src/requests/compat.py:64:4: W0611: Unused import json (unused-import)              │
│        │        │ src/requests/auth.py:143:11: R1714: Consider merging these comparisons with 'in' by │ src/requests/compat.py:67:4: W0611: Unused JSONDecodeError imported from simplejson │
│        │        │ using '_algorithm in ('MD5', 'MD5-SESS')'. Use a set instead if elements are        │ (unused-import)                                                                     │
│        │        │ hashable. (consider-using-in)                                                       │ src/requests/compat.py:72:0: W0611: Unused OrderedDict imported from collections    │
│        │        │ src/requests/auth.py:176:13: C3001: Lambda expression assigned to a variable.       │ (unused-import)                                                                     │
│        │        │ Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)  │ src/requests/compat.py:73:0: W0611: Unused Callable imported from collections.abc   │
│        │        │ src/requests/auth.py:176:13: C3001: Lambda expression assigned to a variable.       │ (unused-import)                                                                     │
│        │        │ Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)  │ src/requests/compat.py:73:0: W0611: Unused Mapping imported from collections.abc    │
│        │        │ src/requests/auth.py:176:8: C0103: Variable name "KD" doesn't conform to snake_case │ (unused-import)                                                                     │
│        │        │ naming style (invalid-name)                                                         │ src/requests/compat.py:73:0: W0611: Unused MutableMapping imported from             │
│        │        │ src/requests/auth.py:189:8: C0103: Variable name "A1" doesn't conform to snake_case │ collections.abc (unused-import)                                                     │
│        │        │ naming style (invalid-name)                                                         │ src/requests/compat.py:74:0: W0611: Unused cookiejar imported from http as          │
│        │        │ src/requests/auth.py:190:8: C0103: Variable name "A2" doesn't conform to snake_case │ cookielib (unused-import)                                                           │
│        │        │ naming style (invalid-name)                                                         │ src/requests/compat.py:75:0: W0611: Unused Morsel imported from http.cookies        │
│        │        │ src/requests/auth.py:192:8: C0103: Variable name "HA1" doesn't conform to           │ (unused-import)                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │ src/requests/compat.py:76:0: W0611: Unused StringIO imported from io                │
│        │        │ src/requests/auth.py:193:8: C0103: Variable name "HA2" doesn't conform to           │ (unused-import)                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │ src/requests/compat.py:81:0: W0611: Unused quote imported from urllib.parse         │
│        │        │ src/requests/auth.py:207:12: C0103: Variable name "HA1" doesn't conform to          │ (unused-import)                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │ src/requests/compat.py:81:0: W0611: Unused quote_plus imported from urllib.parse    │
│        │        │ src/requests/auth.py:126:4: R0912: Too many branches (18/12) (too-many-branches)    │ (unused-import)                                                                     │
│        │        │ src/requests/auth.py:126:4: R0915: Too many statements (66/50)                      │ src/requests/compat.py:81:0: W0611: Unused unquote imported from urllib.parse       │
│        │        │ (too-many-statements)                                                               │ (unused-import)                                                                     │
│        │        │ src/requests/auth.py:126:4: R0912: Too many branches (18/12) (too-many-branches)    │ src/requests/compat.py:81:0: W0611: Unused unquote_plus imported from urllib.parse  │
│        │        │ src/requests/auth.py:126:4: R0915: Too many statements (66/50)                      │ (unused-import)                                                                     │
│        │        │ (too-many-statements)                                                               │ src/requests/compat.py:81:0: W0611: Unused urldefrag imported from urllib.parse     │
│        │        │ src/requests/sessions.py:106:0: C0115: Missing class docstring                      │ (unused-import)                                                                     │
│        │        │ (missing-class-docstring)                                                           │ src/requests/compat.py:81:0: W0611: Unused urlencode imported from urllib.parse     │
│        │        │ src/requests/sessions.py:159:4: R0913: Too many arguments (9/5)                     │ (unused-import)                                                                     │
│        │        │ (too-many-arguments)                                                                │ src/requests/compat.py:81:0: W0611: Unused urljoin imported from urllib.parse       │
│        │        │ src/requests/sessions.py:159:4: R0917: Too many positional arguments (9/5)          │ (unused-import)                                                                     │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/compat.py:81:0: W0611: Unused urlparse imported from urllib.parse      │
│        │        │ src/requests/sessions.py:159:4: R0914: Too many local variables (20/15)             │ (unused-import)                                                                     │
│        │        │ (too-many-locals)                                                                   │ src/requests/compat.py:81:0: W0611: Unused urlsplit imported from urllib.parse      │
│        │        │ src/requests/sessions.py:159:4: R0913: Too many arguments (9/5)                     │ (unused-import)                                                                     │
│        │        │ (too-many-arguments)                                                                │ src/requests/compat.py:81:0: W0611: Unused urlunparse imported from urllib.parse    │
│        │        │ src/requests/sessions.py:159:4: R0917: Too many positional arguments (9/5)          │ (unused-import)                                                                     │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/compat.py:93:0: W0611: Unused getproxies imported from urllib.request  │
│        │        │ src/requests/sessions.py:159:4: R0914: Too many local variables (20/15)             │ (unused-import)                                                                     │
│        │        │ (too-many-locals)                                                                   │ src/requests/compat.py:93:0: W0611: Unused getproxies_environment imported from     │
│        │        │ src/requests/sessions.py:225:16: E1101: Instance of 'LookupDict' has no             │ urllib.request (unused-import)                                                      │
│        │        │ 'temporary_redirect' member (no-member)                                             │ src/requests/compat.py:93:0: W0611: Unused parse_http_list imported from            │
│        │        │ src/requests/sessions.py:225:16: E1101: Instance of 'LookupDict' has no             │ urllib.request (unused-import)                                                      │
│        │        │ 'temporary_redirect' member (no-member)                                             │ src/requests/compat.py:93:0: W0611: Unused proxy_bypass imported from               │
│        │        │ src/requests/sessions.py:226:16: E1101: Instance of 'LookupDict' has no             │ urllib.request (unused-import)                                                      │
│        │        │ 'permanent_redirect' member (no-member)                                             │ src/requests/compat.py:93:0: W0611: Unused proxy_bypass_environment imported from   │
│        │        │ src/requests/sessions.py:226:16: E1101: Instance of 'LookupDict' has no             │ urllib.request (unused-import)                                                      │
│        │        │ 'permanent_redirect' member (no-member)                                             │ src/requests/models.py:41:0: W0622: Redefining built-in 'ConnectionError'           │
│        │        │ src/requests/sessions.py:159:4: R0912: Too many branches (13/12)                    │ (redefined-builtin)                                                                 │
│        │        │ (too-many-branches)                                                                 │ src/requests/models.py:41:0: W0622: Redefining built-in 'ConnectionError'           │
│        │        │ src/requests/sessions.py:159:4: R0915: Too many statements (51/50)                  │ (redefined-builtin)                                                                 │
│        │        │ (too-many-statements)                                                               │ src/requests/models.py:387:8: W0212: Access to a protected member _cookies of a     │
│        │        │ src/requests/sessions.py:159:4: R0912: Too many branches (13/12)                    │ client class (protected-access)                                                     │
│        │        │ (too-many-branches)                                                                 │ src/requests/models.py:390:8: W0212: Access to a protected member _body_position of │
│        │        │ src/requests/sessions.py:159:4: R0915: Too many statements (51/50)                  │ a client class (protected-access)                                                   │
│        │        │ (too-many-statements)                                                               │ src/requests/models.py:406:12: W0707: Consider explicitly re-raising using 'except  │
│        │        │ src/requests/sessions.py:340:35: E1101: Instance of 'LookupDict' has no 'see_other' │ Exception as exc' and 'raise UnicodeError from exc' (raise-missing-from)            │
│        │        │ member (no-member)                                                                  │ src/requests/models.py:406:12: W0707: Consider explicitly re-raising using 'except  │
│        │        │ src/requests/sessions.py:340:35: E1101: Instance of 'LookupDict' has no 'see_other' │ Exception as exc' and 'raise UnicodeError from exc' (raise-missing-from)            │
│        │        │ member (no-member)                                                                  │ src/requests/models.py:435:12: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:345:35: E1101: Instance of 'LookupDict' has no 'found'     │ InvalidURL(*e.args) from e' (raise-missing-from)                                    │
│        │        │ member (no-member)                                                                  │ src/requests/models.py:435:12: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:345:35: E1101: Instance of 'LookupDict' has no 'found'     │ InvalidURL(*e.args) from e' (raise-missing-from)                                    │
│        │        │ member (no-member)                                                                  │ src/requests/models.py:454:16: W0707: Consider explicitly re-raising using 'except  │
│        │        │ src/requests/sessions.py:350:35: E1101: Instance of 'LookupDict' has no 'moved'     │ UnicodeError as exc' and 'raise InvalidURL('URL has an invalid label.') from exc'   │
│        │        │ member (no-member)                                                                  │ (raise-missing-from)                                                                │
│        │        │ src/requests/sessions.py:350:35: E1101: Instance of 'LookupDict' has no 'moved'     │ src/requests/models.py:454:16: W0707: Consider explicitly re-raising using 'except  │
│        │        │ member (no-member)                                                                  │ UnicodeError as exc' and 'raise InvalidURL('URL has an invalid label.') from exc'   │
│        │        │ src/requests/sessions.py:356:0: R0902: Too many instance attributes (12/7)          │ (raise-missing-from)                                                                │
│        │        │ (too-many-instance-attributes)                                                      │ src/requests/models.py:512:16: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:356:0: R0902: Too many instance attributes (12/7)          │ InvalidJSONError(ve, request=self) from ve' (raise-missing-from)                    │
│        │        │ (too-many-instance-attributes)                                                      │ src/requests/models.py:512:16: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:500:4: R0913: Too many arguments (17/5)                    │ InvalidJSONError(ve, request=self) from ve' (raise-missing-from)                    │
│        │        │ (too-many-arguments)                                                                │ src/requests/models.py:588:33: W0613: Unused argument 'url' (unused-argument)       │
│        │        │ src/requests/sessions.py:500:4: R0917: Too many positional arguments (17/5)         │ src/requests/models.py:588:33: W0613: Unused argument 'url' (unused-argument)       │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/models.py:715:12: W0104: Statement seems to have no effect             │
│        │        │ src/requests/sessions.py:500:4: R0914: Too many local variables (22/15)             │ (pointless-statement)                                                               │
│        │        │ (too-many-locals)                                                                   │ src/requests/models.py:822:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:500:4: R0913: Too many arguments (17/5)                    │ ChunkedEncodingError(e) from e' (raise-missing-from)                                │
│        │        │ (too-many-arguments)                                                                │ src/requests/models.py:822:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:500:4: R0917: Too many positional arguments (17/5)         │ ChunkedEncodingError(e) from e' (raise-missing-from)                                │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/models.py:824:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:500:4: R0914: Too many local variables (22/15)             │ ContentDecodingError(e) from e' (raise-missing-from)                                │
│        │        │ (too-many-locals)                                                                   │ src/requests/models.py:824:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:724:22: R1721: Unnecessary use of a comprehension, use     │ ContentDecodingError(e) from e' (raise-missing-from)                                │
│        │        │ list(gen) instead. (unnecessary-comprehension)                                      │ src/requests/models.py:826:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:750:4: R0913: Too many arguments (6/5)                     │ ConnectionError(e) from e' (raise-missing-from)                                     │
│        │        │ (too-many-arguments)                                                                │ src/requests/models.py:826:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:750:4: R0917: Too many positional arguments (6/5)          │ ConnectionError(e) from e' (raise-missing-from)                                     │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/models.py:828:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:750:4: R0913: Too many arguments (6/5)                     │ RequestsSSLError(e) from e' (raise-missing-from)                                    │
│        │        │ (too-many-arguments)                                                                │ src/requests/models.py:828:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/sessions.py:750:4: R0917: Too many positional arguments (6/5)          │ RequestsSSLError(e) from e' (raise-missing-from)                                    │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/models.py:971:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/hooks.py:15:0: C0116: Missing function or method docstring             │ RequestsJSONDecodeError(e.msg, e.doc, e.pos) from e' (raise-missing-from)           │
│        │        │ (missing-function-docstring)                                                        │ src/requests/models.py:971:20: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/compat.py:23:4: C0103: Constant name "is_urllib3_1" doesn't conform to │ RequestsJSONDecodeError(e.msg, e.doc, e.pos) from e' (raise-missing-from)           │
│        │        │ UPPER_CASE naming style (invalid-name)                                              │ src/requests/models.py:978:12: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/compat.py:58:0: C0103: Constant name "has_simplejson" doesn't conform  │ RequestsJSONDecodeError(e.msg, e.doc, e.pos) from e' (raise-missing-from)           │
│        │        │ to UPPER_CASE naming style (invalid-name)                                           │ src/requests/models.py:978:12: W0707: Consider explicitly re-raising using 'raise   │
│        │        │ src/requests/compat.py:62:4: C0103: Constant name "has_simplejson" doesn't conform  │ RequestsJSONDecodeError(e.msg, e.doc, e.pos) from e' (raise-missing-from)           │
│        │        │ to UPPER_CASE naming style (invalid-name)                                           │ src/requests/models.py:13:0: W0611: Unused import encodings.idna (unused-import)    │
│        │        │ src/requests/compat.py:67:4: E0401: Unable to import 'simplejson' (import-error)    │ src/requests/models.py:13:0: W0611: Unused import encodings.idna (unused-import)    │
│        │        │ src/requests/compat.py:67:4: E0401: Unable to import 'simplejson' (import-error)    │ src/requests/status_codes.py:120:4: W0622: Redefining built-in '__doc__'            │
│        │        │ src/requests/compat.py:72:0: C0413: Import "from collections import OrderedDict"    │ (redefined-builtin)                                                                 │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ src/requests/status_codes.py:120:4: W0603: Using the global statement               │
│        │        │ src/requests/compat.py:72:0: C0413: Import "from collections import OrderedDict"    │ (global-statement)                                                                  │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ src/requests/status_codes.py:120:4: W0622: Redefining built-in '__doc__'            │
│        │        │ src/requests/compat.py:73:0: C0413: Import "from collections.abc import Callable,   │ (redefined-builtin)                                                                 │
│        │        │ Mapping, MutableMapping" should be placed at the top of the module                  │ src/requests/status_codes.py:120:4: W0603: Using the global statement               │
│        │        │ (wrong-import-position)                                                             │ (global-statement)                                                                  │
│        │        │ src/requests/compat.py:73:0: C0413: Import "from collections.abc import Callable,   │ src/requests/api.py:73:11: W3101: Missing timeout argument for method 'request' can │
│        │        │ Mapping, MutableMapping" should be placed at the top of the module                  │ cause your program to hang indefinitely (missing-timeout)                           │
│        │        │ (wrong-import-position)                                                             │ src/requests/api.py:73:11: W3101: Missing timeout argument for method 'request' can │
│        │        │ src/requests/compat.py:74:0: C0413: Import "from http import cookiejar as           │ cause your program to hang indefinitely (missing-timeout)                           │
│        │        │ cookielib" should be placed at the top of the module (wrong-import-position)        │ src/requests/api.py:85:11: W3101: Missing timeout argument for method 'request' can │
│        │        │ src/requests/compat.py:74:0: C0413: Import "from http import cookiejar as           │ cause your program to hang indefinitely (missing-timeout)                           │
│        │        │ cookielib" should be placed at the top of the module (wrong-import-position)        │ src/requests/api.py:85:11: W3101: Missing timeout argument for method 'request' can │
│        │        │ src/requests/compat.py:75:0: C0413: Import "from http.cookies import Morsel" should │ cause your program to hang indefinitely (missing-timeout)                           │
│        │        │ be placed at the top of the module (wrong-import-position)                          │ src/requests/api.py💯11: W3101: Missing timeout argument for method 'request' can   │
│        │        │ src/requests/compat.py:75:0: C0413: Import "from http.cookies import Morsel" should │ cause your program to hang indefinitely (missing-timeout)                           │
│        │        │ be placed at the top of the module (wrong-import-position)                          │ src/requests/api.py💯11: W3101: Missing timeout argument for method 'request' can   │
│        │        │ src/requests/compat.py:76:0: C0413: Import "from io import StringIO" should be      │ cause your program to hang indefinitely (missing-timeout)                           │
│        │        │ placed at the top of the module (wrong-import-position)                             │ src/requests/api.py:115:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ src/requests/compat.py:76:0: C0413: Import "from io import StringIO" should be      │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ placed at the top of the module (wrong-import-position)                             │ src/requests/api.py:115:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ src/requests/compat.py:81:0: C0413: Import "from urllib.parse import quote,         │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ quote_plus, unquote, unquote_plus, urldefrag, urlencode, urljoin, urlparse,         │ src/requests/api.py:130:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ urlsplit, urlunparse" should be placed at the top of the module                     │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ (wrong-import-position)                                                             │ src/requests/api.py:130:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ src/requests/compat.py:81:0: C0413: Import "from urllib.parse import quote,         │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ quote_plus, unquote, unquote_plus, urldefrag, urlencode, urljoin, urlparse,         │ src/requests/api.py:145:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ urlsplit, urlunparse" should be placed at the top of the module                     │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ (wrong-import-position)                                                             │ src/requests/api.py:145:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ src/requests/compat.py:93:0: C0413: Import "from urllib.request import getproxies,  │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ getproxies_environment, parse_http_list, proxy_bypass, proxy_bypass_environment"    │ src/requests/api.py:157:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ src/requests/compat.py:93:0: C0413: Import "from urllib.request import getproxies,  │ src/requests/api.py:157:11: W3101: Missing timeout argument for method 'request'    │
│        │        │ getproxies_environment, parse_http_list, proxy_bypass, proxy_bypass_environment"    │ can cause your program to hang indefinitely (missing-timeout)                       │
│        │        │ should be placed at the top of the module (wrong-import-position)                   │ src/requests/utils.py:34:0: W0622: Redefining built-in 'bytes' (redefined-builtin)  │
│        │        │ src/requests/compat.py:101:0: C0103: Class name "builtin_str" doesn't conform to    │ src/requests/utils.py:44:0: W0622: Redefining built-in 'str' (redefined-builtin)    │
│        │        │ PascalCase naming style (invalid-name)                                              │ src/requests/utils.py:34:0: W0622: Redefining built-in 'bytes' (redefined-builtin)  │
│        │        │ src/requests/compat.py:102:0: C0103: Class name "str" doesn't conform to PascalCase │ src/requests/utils.py:44:0: W0622: Redefining built-in 'str' (redefined-builtin)    │
│        │        │ naming style (invalid-name)                                                         │ src/requests/utils.py:652:16: W0707: Consider explicitly re-raising using 'except   │
│        │        │ src/requests/compat.py:103:0: C0103: Class name "bytes" doesn't conform to          │ ValueError as exc' and 'raise InvalidURL(f"Invalid percent-escape sequence: '{h}'") │
│        │        │ PascalCase naming style (invalid-name)                                              │ from exc' (raise-missing-from)                                                      │
│        │        │ src/requests/compat.py:72:0: C0411: standard import "collections.OrderedDict"       │ src/requests/utils.py:652:16: W0707: Consider explicitly re-raising using 'except   │
│        │        │ should be placed before third party import "urllib3.__version__"                    │ ValueError as exc' and 'raise InvalidURL(f"Invalid percent-escape sequence: '{h}'") │
│        │        │ (wrong-import-order)                                                                │ from exc' (raise-missing-from)                                                      │
│        │        │ src/requests/compat.py:73:0: C0411: standard import "collections.abc.Callable"      │ src/requests/utils.py:999:18: W0612: Unused variable 'host' (unused-variable)       │
│        │        │ should be placed before third party import "urllib3.__version__"                    │ src/requests/utils.py:999:24: W0612: Unused variable 'port' (unused-variable)       │
│        │        │ (wrong-import-order)                                                                │ src/requests/utils.py:999:18: W0612: Unused variable 'host' (unused-variable)       │
│        │        │ src/requests/compat.py:74:0: C0411: standard import "http.cookiejar" should be      │ src/requests/utils.py:999:24: W0612: Unused variable 'port' (unused-variable)       │
│        │        │ placed before third party import "urllib3.__version__" (wrong-import-order)         │ src/requests/utils.py:1073:41: W0612: Unused variable 'fragment' (unused-variable)  │
│        │        │ src/requests/compat.py:75:0: C0411: standard import "http.cookies.Morsel" should be │ src/requests/utils.py:1073:41: W0612: Unused variable 'fragment' (unused-variable)  │
│        │        │ placed before third party import "urllib3.__version__" (wrong-import-order)         │ src/requests/utils.py:1090:8: W0212: Access to a protected member _body_position of │
│        │        │ src/requests/compat.py:76:0: C0411: standard import "io.StringIO" should be placed  │ a client class (protected-access)                                                   │
│        │        │ before third party import "urllib3.__version__" (wrong-import-order)                │ src/requests/utils.py:1093:22: W0212: Access to a protected member _body_position   │
│        │        │ src/requests/compat.py:81:0: C0411: standard import "urllib.parse.quote" should be  │ of a client class (protected-access)                                                │
│        │        │ placed before third party import "urllib3.__version__" (wrong-import-order)         │ src/requests/utils.py:1095:12: W0707: Consider explicitly re-raising using 'except  │
│        │        │ src/requests/compat.py:93:0: C0411: standard import "urllib.request.getproxies"     │ OSError as exc' and 'raise UnrewindableBodyError('An error occurred when rewinding  │
│        │        │ should be placed before third party import "urllib3.__version__"                    │ request body for redirect.') from exc' (raise-missing-from)                         │
│        │        │ (wrong-import-order)                                                                │ src/requests/utils.py:1095:12: W0707: Consider explicitly re-raising using 'except  │
│        │        │ src/requests/compat.py:72:0: C0411: standard import "collections.OrderedDict"       │ OSError as exc' and 'raise UnrewindableBodyError('An error occurred when rewinding  │
│        │        │ should be placed before third party import "urllib3.__version__"                    │ request body for redirect.') from exc' (raise-missing-from)                         │
│        │        │ (wrong-import-order)                                                                │ src/requests/utils.py:28:0: W0611: Unused HEADER_VALIDATORS imported from           │
│        │        │ src/requests/compat.py:73:0: C0411: standard import "collections.abc.Callable"      │ _internal_utils (unused-import)                                                     │
│        │        │ should be placed before third party import "urllib3.__version__"                    │ src/requests/utils.py:28:0: W0611: Unused to_native_string imported from            │
│        │        │ (wrong-import-order)                                                                │ _internal_utils (unused-import)                                                     │
│        │        │ src/requests/compat.py:74:0: C0411: standard import "http.cookiejar" should be      │ src/requests/utils.py:28:0: W0611: Unused HEADER_VALIDATORS imported from           │
│        │        │ placed before third party import "urllib3.__version__" (wrong-import-order)         │ _internal_utils (unused-import)                                                     │
│        │        │ src/requests/compat.py:75:0: C0411: standard import "http.cookies.Morsel" should be │ src/requests/utils.py:28:0: W0611: Unused to_native_string imported from            │
│        │        │ placed before third party import "urllib3.__version__" (wrong-import-order)         │ _internal_utils (unused-import)                                                     │
│        │        │ src/requests/compat.py:76:0: C0411: standard import "io.StringIO" should be placed  │ src/requests/exceptions.py:59:0: W0622: Redefining built-in 'ConnectionError'       │
│        │        │ before third party import "urllib3.__version__" (wrong-import-order)                │ (redefined-builtin)                                                                 │
│        │        │ src/requests/compat.py:81:0: C0411: standard import "urllib.parse.quote" should be  │ src/requests/exceptions.py:59:0: W0622: Redefining built-in 'ConnectionError'       │
│        │        │ placed before third party import "urllib3.__version__" (wrong-import-order)         │ (redefined-builtin)                                                                 │
│        │        │ src/requests/compat.py:93:0: C0411: standard import "urllib.request.getproxies"     │ src/requests/adapters.py:686:17: W0511: TODO: Remove this in 3.0.0: see #2811       │
│        │        │ should be placed before third party import "urllib3.__version__"                    │ (fixme)                                                                             │
│        │        │ (wrong-import-order)                                                                │ src/requests/adapters.py:686:17: W0511: TODO: Remove this in 3.0.0: see #2811       │
│        │        │ src/requests/models.py:296:0: C0301: Line too long (102/100) (line-too-long)        │ (fixme)                                                                             │
│        │        │ src/requests/models.py:1:0: C0302: Too many lines in module (1037/1000)             │ src/requests/adapters.py:35:0: W0622: Redefining built-in 'ConnectionError'         │
│        │        │ (too-many-lines)                                                                    │ (redefined-builtin)                                                                 │
│        │        │ src/requests/models.py:296:0: C0301: Line too long (102/100) (line-too-long)        │ src/requests/adapters.py:35:0: W0622: Redefining built-in 'ConnectionError'         │
│        │        │ src/requests/models.py:1:0: C0302: Too many lines in module (1037/1000)             │ (redefined-builtin)                                                                 │
│        │        │ (too-many-lines)                                                                    │ src/requests/adapters.py:140:4: W0246: Useless parent or super() delegation in      │
│        │        │ src/requests/models.py:72:4: E1101: Instance of 'LookupDict' has no 'moved' member  │ method '__init__' (useless-parent-delegation)                                       │
│        │        │ (no-member)                                                                         │ src/requests/adapters.py:474:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:72:4: E1101: Instance of 'LookupDict' has no 'moved' member  │ InvalidURL(e, request=request) from e' (raise-missing-from)                         │
│        │        │ (no-member)                                                                         │ src/requests/adapters.py:474:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:73:4: E1101: Instance of 'LookupDict' has no 'found' member  │ InvalidURL(e, request=request) from e' (raise-missing-from)                         │
│        │        │ (no-member)                                                                         │ src/requests/adapters.py:590:8: W0107: Unnecessary pass statement                   │
│        │        │ src/requests/models.py:73:4: E1101: Instance of 'LookupDict' has no 'found' member  │ (unnecessary-pass)                                                                  │
│        │        │ (no-member)                                                                         │ src/requests/adapters.py:637:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:74:4: E1101: Instance of 'LookupDict' has no 'other' member  │ InvalidURL(e, request=request) from e' (raise-missing-from)                         │
│        │        │ (no-member)                                                                         │ src/requests/adapters.py:637:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:74:4: E1101: Instance of 'LookupDict' has no 'other' member  │ InvalidURL(e, request=request) from e' (raise-missing-from)                         │
│        │        │ (no-member)                                                                         │ src/requests/adapters.py:657:16: W0707: Consider explicitly re-raising using        │
│        │        │ src/requests/models.py:75:4: E1101: Instance of 'LookupDict' has no                 │ 'except ValueError as exc' and 'raise ValueError(f'Invalid timeout {timeout}. Pass  │
│        │        │ 'temporary_redirect' member (no-member)                                             │ a (connect, read) timeout tuple, or a single float to set both timeouts to the same │
│        │        │ src/requests/models.py:75:4: E1101: Instance of 'LookupDict' has no                 │ value.') from exc' (raise-missing-from)                                             │
│        │        │ 'temporary_redirect' member (no-member)                                             │ src/requests/adapters.py:657:16: W0707: Consider explicitly re-raising using        │
│        │        │ src/requests/models.py:76:4: E1101: Instance of 'LookupDict' has no                 │ 'except ValueError as exc' and 'raise ValueError(f'Invalid timeout {timeout}. Pass  │
│        │        │ 'permanent_redirect' member (no-member)                                             │ a (connect, read) timeout tuple, or a single float to set both timeouts to the same │
│        │        │ src/requests/models.py:76:4: E1101: Instance of 'LookupDict' has no                 │ value.') from exc' (raise-missing-from)                                             │
│        │        │ 'permanent_redirect' member (no-member)                                             │ src/requests/adapters.py:682:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:84:0: C0115: Missing class docstring                         │ ConnectionError(err, request=request) from err' (raise-missing-from)                │
│        │        │ (missing-class-docstring)                                                           │ src/requests/adapters.py:682:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:115:8: R1705: Unnecessary "elif" after "return", remove the  │ ConnectionError(err, request=request) from err' (raise-missing-from)                │
│        │        │ leading "el" from "elif" (no-else-return)                                           │ src/requests/adapters.py:688:20: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:137:4: R0914: Too many local variables (16/15)               │ ConnectTimeout(e, request=request) from e' (raise-missing-from)                     │
│        │        │ (too-many-locals)                                                                   │ src/requests/adapters.py:688:20: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:137:4: R0914: Too many local variables (16/15)               │ ConnectTimeout(e, request=request) from e' (raise-missing-from)                     │
│        │        │ (too-many-locals)                                                                   │ src/requests/adapters.py:691:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:146:8: R1720: Unnecessary "elif" after "raise", remove the   │ RetryError(e, request=request) from e' (raise-missing-from)                         │
│        │        │ leading "el" from "elif" (no-else-raise)                                            │ src/requests/adapters.py:691:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:137:4: R0912: Too many branches (17/12) (too-many-branches)  │ RetryError(e, request=request) from e' (raise-missing-from)                         │
│        │        │ src/requests/models.py:137:4: R0912: Too many branches (17/12) (too-many-branches)  │ src/requests/adapters.py:694:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:84:0: R0903: Too few public methods (1/2)                    │ ProxyError(e, request=request) from e' (raise-missing-from)                         │
│        │        │ (too-few-public-methods)                                                            │ src/requests/adapters.py:694:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:84:0: R0903: Too few public methods (1/2)                    │ ProxyError(e, request=request) from e' (raise-missing-from)                         │
│        │        │ (too-few-public-methods)                                                            │ src/requests/adapters.py:698:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:206:0: C0115: Missing class docstring                        │ SSLError(e, request=request) from e' (raise-missing-from)                           │
│        │        │ (missing-class-docstring)                                                           │ src/requests/adapters.py:698:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:230:0: R0902: Too many instance attributes (10/7)            │ SSLError(e, request=request) from e' (raise-missing-from)                           │
│        │        │ (too-many-instance-attributes)                                                      │ src/requests/adapters.py:700:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:230:0: R0902: Too many instance attributes (10/7)            │ ConnectionError(e, request=request) from e' (raise-missing-from)                    │
│        │        │ (too-many-instance-attributes)                                                      │ src/requests/adapters.py:700:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:258:4: R0913: Too many arguments (11/5) (too-many-arguments) │ ConnectionError(e, request=request) from e' (raise-missing-from)                    │
│        │        │ src/requests/models.py:258:4: R0917: Too many positional arguments (11/5)           │ src/requests/adapters.py:703:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ (too-many-positional-arguments)                                                     │ ConnectionError(e, request=request) from e' (raise-missing-from)                    │
│        │        │ src/requests/models.py:258:4: R0913: Too many arguments (11/5) (too-many-arguments) │ src/requests/adapters.py:703:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:258:4: R0917: Too many positional arguments (11/5)           │ ConnectionError(e, request=request) from e' (raise-missing-from)                    │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/adapters.py:706:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:351:4: R0913: Too many arguments (11/5) (too-many-arguments) │ ProxyError(e) from e' (raise-missing-from)                                          │
│        │        │ src/requests/models.py:351:4: R0917: Too many positional arguments (11/5)           │ src/requests/adapters.py:706:12: W0707: Consider explicitly re-raising using 'raise │
│        │        │ (too-many-positional-arguments)                                                     │ ProxyError(e) from e' (raise-missing-from)                                          │
│        │        │ src/requests/models.py:351:4: R0913: Too many arguments (11/5) (too-many-arguments) │ src/requests/adapters.py:711:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:351:4: R0917: Too many positional arguments (11/5)           │ SSLError(e, request=request) from e' (raise-missing-from)                           │
│        │        │ (too-many-positional-arguments)                                                     │ src/requests/adapters.py:711:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:382:4: C0116: Missing function or method docstring           │ SSLError(e, request=request) from e' (raise-missing-from)                           │
│        │        │ (missing-function-docstring)                                                        │ src/requests/adapters.py:713:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:401:8: C0415: Import outside toplevel (idna)                 │ ReadTimeout(e, request=request) from e' (raise-missing-from)                        │
│        │        │ (import-outside-toplevel)                                                           │ src/requests/adapters.py:713:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:401:8: C0415: Import outside toplevel (idna)                 │ ReadTimeout(e, request=request) from e' (raise-missing-from)                        │
│        │        │ (import-outside-toplevel)                                                           │ src/requests/adapters.py:715:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:409:4: R0912: Too many branches (16/12) (too-many-branches)  │ InvalidHeader(e, request=request) from e' (raise-missing-from)                      │
│        │        │ src/requests/models.py:409:4: R0912: Too many branches (16/12) (too-many-branches)  │ src/requests/adapters.py:715:16: W0707: Consider explicitly re-raising using 'raise │
│        │        │ src/requests/models.py:494:4: R0912: Too many branches (16/12) (too-many-branches)  │ InvalidHeader(e, request=request) from e' (raise-missing-from)                      │
│        │        │ src/requests/models.py:494:4: R0912: Too many branches (16/12) (too-many-branches)  │ src/requests/adapters.py:10:0: W0611: Unused import socket (unused-import)          │
│        │        │ src/requests/models.py:640:0: R0902: Too many instance attributes (13/7)            │ src/requests/adapters.py:78:4: W0611: Unused import ssl (unused-import)             │
│        │        │ (too-many-instance-attributes)                                                      │ src/requests/adapters.py:10:0: W0611: Unused import socket (unused-import)          │
│        │        │ src/requests/models.py:640:0: R0902: Too many instance attributes (13/7)            │ src/requests/adapters.py:78:4: W0611: Unused import ssl (unused-import)             │
│        │        │ (too-many-instance-attributes)                                                      │                                                                                     │
│        │        │ src/requests/models.py:780:12: E1101: Instance of 'LookupDict' has no               │                                                                                     │
│        │        │ 'moved_permanently' member (no-member)                                              │                                                                                     │
│        │        │ src/requests/models.py:780:12: E1101: Instance of 'LookupDict' has no               │                                                                                     │
│        │        │ 'moved_permanently' member (no-member)                                              │                                                                                     │
│        │        │ src/requests/models.py:781:12: E1101: Instance of 'LookupDict' has no               │                                                                                     │
│        │        │ 'permanent_redirect' member (no-member)                                             │                                                                                     │
│        │        │ src/requests/models.py:781:12: E1101: Instance of 'LookupDict' has no               │                                                                                     │
│        │        │ 'permanent_redirect' member (no-member)                                             │                                                                                     │
│        │        │ src/requests/models.py:792:8: R1705: Unnecessary "else" after "return", remove the  │                                                                                     │
│        │        │ "else" and de-indent the code inside it (no-else-return)                            │                                                                                     │
│        │        │ src/requests/models.py:839:8: R1720: Unnecessary "elif" after "raise", remove the   │                                                                                     │
│        │        │ leading "el" from "elif" (no-else-raise)                                            │                                                                                     │
│        │        │ src/requests/status_codes.py:118:15: C0209: Formatting a regular string which could │                                                                                     │
│        │        │ be an f-string (consider-using-f-string)                                            │                                                                                     │
│        │        │ src/requests/packages.py:1:0: C0114: Missing module docstring                       │                                                                                     │
│        │        │ (missing-module-docstring)                                                          │                                                                                     │
│        │        │ src/requests/__version__.py:1:0: C0114: Missing module docstring                    │                                                                                     │
│        │        │ (missing-module-docstring)                                                          │                                                                                     │
│        │        │ src/requests/api.py:17:0: C0301: Line too long (139/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:23:0: C0301: Line too long (106/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:26:0: C0301: Line too long (133/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:27:0: C0301: Line too long (116/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:28:0: C0301: Line too long (116/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:29:0: C0301: Line too long (123/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:36:0: C0301: Line too long (136/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:43:0: C0301: Line too long (107/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:109:0: C0301: Line too long (106/100) (line-too-long)           │                                                                                     │
│        │        │ src/requests/api.py:124:0: C0301: Line too long (106/100) (line-too-long)           │                                                                                     │
│        │        │ src/requests/api.py:139:0: C0301: Line too long (106/100) (line-too-long)           │                                                                                     │
│        │        │ src/requests/api.py:17:0: C0301: Line too long (139/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:23:0: C0301: Line too long (106/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:26:0: C0301: Line too long (133/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:27:0: C0301: Line too long (116/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:28:0: C0301: Line too long (116/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:29:0: C0301: Line too long (123/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:36:0: C0301: Line too long (136/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:43:0: C0301: Line too long (107/100) (line-too-long)            │                                                                                     │
│        │        │ src/requests/api.py:109:0: C0301: Line too long (106/100) (line-too-long)           │                                                                                     │
│        │        │ src/requests/api.py:124:0: C0301: Line too long (106/100) (line-too-long)           │                                                                                     │
│        │        │ src/requests/api.py:139:0: C0301: Line too long (106/100) (line-too-long)           │                                                                                     │
│        │        │ src/requests/utils.py:286:0: C0301: Line too long (128/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:287:0: C0301: Line too long (105/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:302:0: C0301: Line too long (117/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:563:0: C0301: Line too long (106/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:920:0: C0301: Line too long (118/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:1:0: C0302: Too many lines in module (1099/1000)              │                                                                                     │
│        │        │ (too-many-lines)                                                                    │                                                                                     │
│        │        │ src/requests/utils.py:286:0: C0301: Line too long (128/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:287:0: C0301: Line too long (105/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:302:0: C0301: Line too long (117/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:563:0: C0301: Line too long (106/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:920:0: C0301: Line too long (118/100) (line-too-long)         │                                                                                     │
│        │        │ src/requests/utils.py:1:0: C0302: Too many lines in module (1099/1000)              │                                                                                     │
│        │        │ (too-many-lines)                                                                    │                                                                                     │
│        │        │ src/requests/utils.py:77:4: C0116: Missing function or method docstring             │                                                                                     │
│        │        │ (missing-function-docstring)                                                        │                                                                                     │
│        │        │ src/requests/utils.py:79:12: C0415: Import outside toplevel (winreg)                │                                                                                     │
│        │        │ (import-outside-toplevel)                                                           │                                                                                     │
│        │        │ src/requests/utils.py:79:12: C0415: Import outside toplevel (winreg)                │                                                                                     │
│        │        │ (import-outside-toplevel)                                                           │                                                                                     │
│        │        │ src/requests/utils.py:84:12: C0103: Variable name "internetSettings" doesn't        │                                                                                     │
│        │        │ conform to snake_case naming style (invalid-name)                                   │                                                                                     │
│        │        │ src/requests/utils.py:89:12: C0103: Variable name "proxyEnable" doesn't conform to  │                                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │                                                                                     │
│        │        │ src/requests/utils.py:91:12: C0103: Variable name "proxyOverride" doesn't conform   │                                                                                     │
│        │        │ to snake_case naming style (invalid-name)                                           │                                                                                     │
│        │        │ src/requests/utils.py💯8: C0103: Variable name "proxyOverride" doesn't conform to   │                                                                                     │
│        │        │ snake_case naming style (invalid-name)                                              │                                                                                     │
│        │        │ src/requests/utils.py:102:8: C0103: Variable name "proxyOverride" doesn't conform   │                                                                                     │
│        │        │ to snake_case naming style (invalid-name)                                           │                                                                                     │
│        │        │ src/requests/utils.py:115:4: E0102: function already defined line 44                │                                                                                     │
│        │        │ (function-redefined)                                                                │                                                                                     │
│        │        │ src/requests/utils.py:121:8: R1705: Unnecessary "else" after "return", remove the   │                                                                                     │
│        │        │ "else" and de-indent the code inside it (no-else-return)                            │                                                                                     │
│        │        │ src/requests/utils.py:136:0: C0116: Missing function or method docstring            │                                                                                     │
│        │        │ (missing-function-docstring)                                                        │                                                                                     │
│        │        │ src/requests/utils.py:136:0: R0912: Too many branches (14/12) (too-many-branches)   │                                                                                     │
│        │        │ src/requests/utils.py:136:0: R0912: Too many branches (14/12) (too-many-branches)   │                                                                                     │
│        │        │ src/requests/utils.py:217:8: C0415: Import outside toplevel (netrc.NetrcParseError, │                                                                                     │
│        │        │ netrc.netrc) (import-outside-toplevel)                                              │                                                                                     │
│        │        │ src/requests/utils.py:217:8: C0415: Import outside toplevel (netrc.NetrcParseError, │                                                                                     │
│        │        │ netrc.netrc) (import-outside-toplevel)                                              │                                                                                     │
│        │        │ src/requests/utils.py:207:0: R1710: Either all return statements in a function      │                                                                                     │
│        │        │ should return an expression, or none of them should.                                │                                                                                     │
│        │        │ (inconsistent-return-statements)                                                    │                                                                                     │
│        │        │ src/requests/utils.py:264:0: R1710: Either all return statements in a function      │                                                                                     │
│        │        │ should return an expression, or none of them should.                                │                                                                                     │
│        │        │ (inconsistent-return-statements)                                                    │                                                                                     │
│        │        │ src/requests/utils.py:294:15: R1732: Consider using 'with' for resource-allocating  │                                                                                     │
│        │        │ operations (consider-using-with)                                                    │                                                                                     │
│        │        │ src/requests/utils.py:542:0: R1710: Either all return statements in a function      │                                                                                     │
│        │        │ should return an expression, or none of them should.                                │                                                                                     │
│        │        │ (inconsistent-return-statements)                                                    │                                                                                     │
│        │        │ src/requests/utils.py:768:0: R0912: Too many branches (14/12) (too-many-branches)   │                                                                                     │
│        │        │ src/requests/utils.py:768:0: R0912: Too many branches (14/12) (too-many-branches)   │                                                                                     │
│        │        │ src/requests/utils.py:835:4: R1705: Unnecessary "else" after "return", remove the   │                                                                                     │
│        │        │ "else" and de-indent the code inside it (no-else-return)                            │                                                                                     │
│        │        │ src/requests/utils.py:955:0: C0103: Constant name "_null" doesn't conform to        │                                                                                     │
│        │        │ UPPER_CASE naming style (invalid-name)                                              │                                                                                     │
│        │        │ src/requests/utils.py:956:0: C0103: Constant name "_null2" doesn't conform to       │                                                                                     │
│        │        │ UPPER_CASE naming style (invalid-name)                                              │                                                                                     │
│        │        │ src/requests/utils.py:957:0: C0103: Constant name "_null3" doesn't conform to       │                                                                                     │
│        │        │ UPPER_CASE naming style (invalid-name)                                              │                                                                                     │
│        │        │ src/requests/utils.py:960:0: R0911: Too many return statements (9/6)                │                                                                                     │
│        │        │ (too-many-return-statements)                                                        │                                                                                     │
│        │        │ src/requests/utils.py:960:0: R0911: Too many return statements (9/6)                │                                                                                     │
│        │        │ (too-many-return-statements)                                                        │                                                                                     │
│        │        │ src/requests/structures.py:76:4: C0116: Missing function or method docstring        │                                                                                     │
│        │        │ (missing-function-docstring)                                                        │                                                                                     │
│        │        │ src/requests/help.py:50:33: C0209: Formatting a regular string which could be an    │                                                                                     │
│        │        │ f-string (consider-using-f-string)                                                  │                                                                                     │
│        │        │ src/requests/help.py:51:12: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:51:12: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:52:12: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:52:12: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:53:12: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:53:12: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:55:11: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:55:11: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:57:41: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/help.py:57:41: E1101: Module 'sys' has no 'pypy_version_info' member   │                                                                                     │
│        │        │ (no-member)                                                                         │                                                                                     │
│        │        │ src/requests/adapters.py:63:4: C0116: Missing function or method docstring          │                                                                                     │
│        │        │ (missing-function-docstring)                                                        │                                                                                     │
│        │        │ src/requests/adapters.py:63:4: C0103: Function name "SOCKSProxyManager" doesn't     │                                                                                     │
│        │        │ conform to snake_case naming style (invalid-name)                                   │                                                                                     │
│        │        │ src/requests/adapters.py:87:4: C0103: Constant name "_preloaded_ssl_context"        │                                                                                     │
│        │        │ doesn't conform to UPPER_CASE naming style (invalid-name)                           │                                                                                     │
│        │        │ src/requests/adapters.py:143:4: R0913: Too many arguments (7/5)                     │                                                                                     │
│        │        │ (too-many-arguments)                                                                │                                                                                     │
│        │        │ src/requests/adapters.py:143:4: R0917: Too many positional arguments (7/5)          │                                                                                     │
│        │        │ (too-many-positional-arguments)                                                     │                                                                                     │
│        │        │ src/requests/adapters.py:143:4: R0913: Too many arguments (7/5)                     │                                                                                     │
│        │        │ (too-many-arguments)                                                                │                                                                                     │
│        │        │ src/requests/adapters.py:143:4: R0917: Too many positional arguments (7/5)          │                                                                                     │
│        │        │ (too-many-positional-arguments)                                                     │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0913: Too many arguments (7/5)                     │                                                                                     │
│        │        │ (too-many-arguments)                                                                │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0917: Too many positional arguments (7/5)          │                                                                                     │
│        │        │ (too-many-positional-arguments)                                                     │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0913: Too many arguments (7/5)                     │                                                                                     │
│        │        │ (too-many-arguments)                                                                │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0917: Too many positional arguments (7/5)          │                                                                                     │
│        │        │ (too-many-positional-arguments)                                                     │                                                                                     │
│        │        │ src/requests/adapters.py:709:12: R1720: Unnecessary "elif" after "raise", remove    │                                                                                     │
│        │        │ the leading "el" from "elif" (no-else-raise)                                        │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0912: Too many branches (19/12)                    │                                                                                     │
│        │        │ (too-many-branches)                                                                 │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0915: Too many statements (51/50)                  │                                                                                     │
│        │        │ (too-many-statements)                                                               │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0912: Too many branches (19/12)                    │                                                                                     │
│        │        │ (too-many-branches)                                                                 │                                                                                     │
│        │        │ src/requests/adapters.py:613:4: R0915: Too many statements (51/50)                  │                                                                                     │
│        │        │ (too-many-statements)                                                               │                                                                                     │
│        │        │ src/requests/adapters.py:1:0: R0801: Similar lines in 2 files                       │                                                                                     │
│        │        │ src/requests/adapters.py:1:0: R0801: Similar lines in 2 files                       │                                                                                     │
└────────┴────────┴─────────────────────────────────────────────────────────────────────────────────────┴─────────────────────────────────────────────────────────────────────────────────────┘
Error: Overall Score (89.9) is below the threshold (100.0).
